### PR TITLE
docs/fix: update aria pattern links to point to new W3 APG site

### PR DESCRIPTION
**Pull request by hbuchel**
The [WAI Aria Practices](https://www.w3.org/TR/wai-aria-practices/#abstract) site we were using to link to our component's aria patterns has been deprecated in favor of the new [Aria Authoring Practices Guide](https://www.w3.org/WAI/ARIA/apg/). This PR updates all of our links, example:

```
ariaPattern: https://www.w3.org/WAI/ARIA/apg/patterns/button
```

#### Description of how you validated changes

Tested in docs instance locally and verified each new link works.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/pages/[platform]/components/grid/index.page.mdx [hunk 1]
```diff
@@ -3,7 +3,6 @@ title: Grid
 metaTitle: Grid
 description: Grid provides a CSS grid container.
 metaDescription: Grid provides a CSS grid container.
-ariaPattern: https://www.w3.org/TR/wai-aria-practices-1.2/#grid
```
Comment (hbuchel) [inline docs/src/pages/[platform]/components/grid/index.page.mdx:6]: Our grid component is presentational, not an ARIA grid (which has expectations of directional navigation, focusable content in each "cell", etc) so I removed it.

Comment (slaymance) [reply docs/src/pages/[platform]/components/grid/index.page.mdx:4]: Good catch. Do you think we should roadmap this to bring the component more inline with the ARIA spec?

Comment (hbuchel) [reply docs/src/pages/[platform]/components/grid/index.page.mdx:4]: Hmm no, maybe as a separate, new component (like for creating an [editable spreadsheet](https://www.w3.org/WAI/ARIA/apg/example-index/grid/dataGrids.html#ex2_label) maybe?) but different from this purely CSS layout component.

**Review by reesscot — APPROVED** [review]
Thanks for catching this!

**Pull request merged**